wordpress-mobile/WordPress-iOS
# Update code freeze lane

Lately we have been experimenting with a different code freeze flow where we update the beta pods after the release branch has been cut. 
This PR updates the `code_freeze` lane to reflect this flow:
- the `localize` and `tag` steps are removed from the `code_freeze` lane and moved to the newly created `complete_code_freeze` one.
- the `ios_check_beta_deps` action has been added. It searches for beta pods in the `Podfile` and shows the list and a prompt to release them. 

The related changes in `release-toolkit` are here: https://github.com/wordpress-mobile/release-toolkit/pull/119.

To test:
**A. New code_freeze lane**
_Set up:_
1. On your local machine, add a `14.8` tag. 

_Test:_ 
2. Run `bundle install` and `bundle exec fastlane code_freeze`.
3. Verify that the lane completes successfully and that the list of beta pods is shown in the output. 

_Wrap up:_
4. Delete the new `release/14.9` branch from your computer. 
5. Remove the _frozen_ tag from the `14.9` milestone on GitHub.
6. Remove the branch protection for the `release/14.9` branch on GitHub.
7. Delete the `release/14.9` branch from GitHub.

**B. complete_code_freeze lane**
8. Run `bundle exec complete_code_freeze` lane. Verify that it fails because of the branch name. 
9. Check out a new `release/14.9` branch on your local machine from this branch.
10. Run `bundle exec complete_code_freeze` lane. Verify that it prompts for completion. Answer `No`.

**Note:** The only new code of the `complete_code_freeze` lane is the `prechecks` action that you tested above. The other actions have just been moved, so there's no need to retest them.
Anyway, if you want to test the whole lane, please remember that the last action in the lane will run a `Release Build` job on CircleCI. Please, make sure to delete it job on CircleCI before it completes, or it will upload a new beta to TestFlight that can't be deleted. 

PR submission checklist:
